appservice/src/main/java/nu/yona/server/subscriptions/rest/UserController.java

```diff
@@ -405,15 +472,66 @@ private void setSslRootCertificateCn()
 		}
 	}
 
+	static class PostPutUserDto
+	{
+		private final String firstName;
+		private final String lastName;
+		private final String mobileNumber;
+		private final String nickname;
+		private final String deviceName;
+		private final String deviceOperatingSystem;
+
+		@JsonCreator
+		public PostPutUserDto(@JsonProperty("firstName") String firstName, @JsonProperty("lastName") String lastName,
+				@JsonProperty("mobileNumber") String mobileNumber, @JsonProperty("nickname") String nickname,
+				@JsonProperty(value = "deviceName", required = false) String deviceName,
+				@JsonProperty(value = "deviceOperatingSystem", required = false) String deviceOperatingSystem,
+				@JsonProperty("_links") Object ignored1, @JsonProperty("yonaPassword") Object ignored2)
+		{
+			this.firstName = firstName;
+			this.lastName = lastName;
+			this.mobileNumber = mobileNumber;
+			this.nickname = nickname;
+			this.deviceName = deviceName;
+			this.deviceOperatingSystem = deviceOperatingSystem;
+		}
+
+		Optional<UserDeviceDto> getDevice()
+		{
+			return (deviceName == null) ? Optional.empty()
+					: Optional.of(UserDeviceDto.createPostPutInstance(deviceName, deviceOperatingSystem));
+		}
+	}
+
+	enum Purpose
```

Comment:
This enum was introduced to have an overview of all states of a user DTO, isn't it? I think a better name for it may be `UserResourcePresentationState`.

Reply:
The enum was created inside the {{UserResourceAssembler}}. There {{Purpose}} was quite OK, but due to a compiler bug, I had to move it out of that class, making the name less meaningful. I'll rename it into ``UserResourceRepresentation``. Hope that's fine.

Reply:
Perfect